### PR TITLE
LSIF: Remove noisy NoLSIFDumpError

### PR DESCRIPTION
**Pull request by efritz**
Throwing `NoLSIFDumpError` causes instrumented blocks to emit the exception. This makes it very noisy (and alarming) in prod where there are no actual errors, but the /exists endpoint working normally.

This returns undefined database/dump in place of a thrown exception. This has the (intended) consequence of emitting a warning _only_ in places that don't expect such a condition (the endpoints that are meant to be called after checking if code intelligence data exists).

A secondary consequence changes the behavior of /request endpoints by returning a 200 when no intelligence data is found where it used to return a 500. I think this is an improvement, but we could do even better with a specialized error message or status code to indicate that the request was not called with the correct ordering (or was called in the presence of a race condition such as a dump being deleted).

### CODE REVIEW

**Comment by chrismwendt** [comment]
> A secondary consequence changes the behavior of /request endpoints by returning a 200 when no intelligence data is found where it used to return a 500. I think this is an improvement, but we could do even better with a specialized error message or status code to indicate that the request was not called with the correct ordering (or was called in the presence of a race condition such as a dump being deleted).

Would a 404 be appropriate in this case?

File: lsif/src/server/backend/backend.ts [hunk 1]
```diff
@@ -704,29 +711,27 @@ export class Backend {
         commit: string,
         file: string,
         ctx: TracingContext = {}
-    ): Promise<{ database: Database; dump: xrepoModels.LsifDump; ctx: TracingContext }> {
+    ): Promise<{ database?: Database; dump?: xrepoModels.LsifDump; ctx?: TracingContext }> {
```
Comment (chrismwendt) [inline lsif/src/server/backend/backend.ts]:
```suggestion
    ): Promise<{ database: Database; dump: xrepoModels.LsifDump; ctx: TracingContext } | undefined> {
```

Comment (efritz) [reply lsif/src/server/backend/backend.ts]: Updated.

**Comment by efritz** [comment]
> Would a 404 be appropriate in this case?

I think so. Updated.

**Pull request merged**